pygis/docs/e_extraction.md

```diff
@@ -0,0 +1,350 @@
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply with Depth Greater than 50 ft or Residential", attribute_table = wells_subset_3)
+```
+
+## Select by Location
+
+Selecting by location selects features based on its relative spatial relationship with another dataset.
+
+There are multiple spatial relationships available in `geopandas`.
+
+| Spatial Relationship | Function(s) |
+| :------------ | ---------------------------------------: |
+| contains | `contains()` |
+| covers | `covers()` |
+| covered by | `covered_by()` |
+| crosses | `crosses()` |
+| disjoint | `disjoint()` |
+| equal geometry | `geom_equals()`, `geom_almost_equals()`, `geom_equals_exact()` |
+| intersects | `intersects()` |
+| overlaps | `overlaps()` |
+| touches | `touches()` |
+| within | `within()` |
+
+Source: [GeoSeries - Binary Predicates, GeoPandas](https://geopandas.org/docs/reference/geoseries.html#binary-predicates)
+
+The functions for these spatial relationships will generally output a `pandas` series with Boolean values (`True` or `False`) whose indexing corresponds with the input dataset (from where we want to subset the data). We can use these Boolean values to subset the dataset (only the rows that have a `True` output will be retained).
+
+These functions can be used to select features that have the specified spatial relationship with a single Shapely vector.
+
+```{code-cell} ipython3
+# Select wells that fall within Shapely rectangle
+wells_location_subset_1 = wells[wells.within(poly_shapely)]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells within a User-Defined Rectangle", attribute_table = wells_location_subset_1)
+
+# Plot selection
+plot_df(result_name = "San Francisco Bay Area Wells within a User-Defined Rectangle", result_df = wells_location_subset_1, result_geom_type = "point", area = poly)
+```
+
+If we're trying to select features that have a specified spatial relationship with another `geopandas` object, it gets a little tricky. This is because the `geopandas` spatial relationship functions verify the spatial relationship either row by row or index by index. In other words, the first row in the first dataset will be compared with the corresponding row or index in the second dataset, the second row in the first dataset will be compared with the corresponding row or index in the second dataset, and so on.
+
+As a result, the number of rows need to correspond or the indices numbers need to match between the two datasets--or else we'll get a warning and the output will be empty.
+
+Not easy if we want to check a bunch of features against one extent (with one geometry)... Fortunately, there are two workarounds that get the same result.
+
+Source: [geopandas.GeoSeries.within, GeoPandas](https://geopandas.org/docs/reference/api/geopandas.GeoSeries.within.html)
+
+### Workaround 1 - Manipulate extent GeoDataFrame
```

Comment:
this is great didn't know about this

Reply:
I haven't done much with these funcitons in python but I remember it being tricky and being confused on why things did and didn't work.  I do however want to make sure these are the simplest answers to this. For instance, wouldn't it also be possible to just grab the geom column value (which is stored as a shapely object i think), and comparing to that? So something like `wells[wells.within(sc_county.geometry)]`

Reply:
good point, that's an easier method